digraph "CFG forkotlinx.coroutines.CancellableContinuationImpl.callCancelHandlerSafely(Lkotlin\/jvm\/functions\/Function0;)V" {
Node_11 [shape=record,label="{11\:\ ?|MTH_ENTER_BLOCK\l}"];
Node_13 [shape=record,label="{13\:\ 0x0000|0x0000: CONST  (r0 I:??[int, float, boolean, short, byte, char, OBJECT, ARRAY]) = (0 ??[int, float, boolean, short, byte, char, OBJECT, ARRAY]) \l}"];
Node_19 [shape=record,label="{19\:\ 0x0002|SYNTHETIC\lEXC_TOP_SPLITTER\l}"];
Node_14 [shape=record,label="{14\:\ 0x0002|TRY_ENTER\lTRY_LEAVE\lTryCatch #0 \{all \-\> 0x0006, blocks: (B:14:0x0002), top: B:19:0x0002 \}\lCatch: all \-\> 0x0006\l|0x0002: INVOKE  (r6 I:kotlin.jvm.functions.Function0 A[D('block' kotlin.jvm.functions.Function0)]) type: INTERFACE call: kotlin.jvm.functions.Function0.invoke():java.lang.Object A[Catch: all \-\> 0x0006, MD:():R (m), TRY_ENTER, TRY_LEAVE] (LINE:197)\l}"];
Node_20 [shape=record,label="{20\:\ ?|SYNTHETIC\lEXC_BOTTOM_SPLITTER\l}"];
Node_21 [shape=record,label="{21\:\ ?|SYNTHETIC\lRETURN\l|0x001b: RETURN   A[SYNTHETIC] (LINE:205)\l}"];
Node_12 [shape=record,label="{12\:\ ?|MTH_EXIT_BLOCK\l}"];
Node_16 [shape=record,label="{16\:\ 0x0006|ExcHandler: all \-\> 0x0006\l|0x0006: MOVE_EXCEPTION  (r1 I:java.lang.Throwable A[CUSTOM_DECLARE]) =  A[DONT_INLINE, ExcHandler: all \-\> 0x0006] (LINE:198)\l}"];
Node_17 [shape=record,label="{17\:\ 0x0007|0x0007: INVOKE  (r2 I:kotlin.coroutines.CoroutineContext) = (r5 I:kotlinx.coroutines.CancellableContinuationImpl) type: VIRTUAL call: kotlinx.coroutines.CancellableContinuationImpl.getContext():kotlin.coroutines.CoroutineContext A[MD:():kotlin.coroutines.CoroutineContext (m)] (LINE:201)\l0x000b: NEW_INSTANCE  (r3 I:kotlinx.coroutines.CompletionHandlerException) =  kotlinx.coroutines.CompletionHandlerException  (LINE:202)\l0x000d: CONST_STR  (r4 I:java.lang.String) =  \"Exception in invokeOnCancellation handler for \" \l0x000f: INVOKE  (r4 I:java.lang.String) = (r4 I:java.lang.String), (r5 I:java.lang.Object) type: STATIC call: kotlin.jvm.internal.Intrinsics.stringPlus(java.lang.String, java.lang.Object):java.lang.String A[MD:(java.lang.String, java.lang.Object):java.lang.String (m)]\l0x0013: INVOKE  \l  (r3 I:kotlinx.coroutines.CompletionHandlerException)\l  (r4 I:java.lang.String)\l  (r1 I:java.lang.Throwable A[D('ex' java.lang.Throwable)])\l type: DIRECT call: kotlinx.coroutines.CompletionHandlerException.\<init\>(java.lang.String, java.lang.Throwable):void A[MD:(java.lang.String, java.lang.Throwable):void (m)]\l0x0016: CHECK_CAST (r3 I:java.lang.Throwable) = (java.lang.Throwable) (r3 I:??[OBJECT, ARRAY]) \l0x0018: INVOKE  (r2 I:kotlin.coroutines.CoroutineContext), (r3 I:java.lang.Throwable) type: STATIC call: kotlinx.coroutines.CoroutineExceptionHandlerKt.handleCoroutineException(kotlin.coroutines.CoroutineContext, java.lang.Throwable):void A[MD:(kotlin.coroutines.CoroutineContext, java.lang.Throwable):void (m)] (LINE:200)\l}"];
Node_18 [shape=record,label="{18\:\ 0x001b|RETURN\lORIG_RETURN\l|0x001b: RETURN    (LINE:205)\l}"];
MethodNode[shape=record,label="{private final void kotlinx.coroutines.CancellableContinuationImpl.callCancelHandlerSafely((r5 'this' I:kotlinx.coroutines.CancellableContinuationImpl\<T\> A[IMMUTABLE_TYPE, THIS]), (r6 I:kotlin.jvm.functions.Function0\<kotlin.Unit\> A[D('block' kotlin.jvm.functions.Function0), IMMUTABLE_TYPE, METHOD_ARGUMENT]))  | USE_LINES_HINTS\lCROSS_EDGE: B:17:0x0007 \-\> B:18:0x001b, CROSS_EDGE: B:19:0x0002 \-\> B:16:0x0006\lINLINE_NOT_NEEDED\lTryCatch #0 \{all \-\> 0x0006, blocks: (B:14:0x0002), top: B:19:0x0002 \}\lTYPE_VARS: [T]\lSIGNATURE: (Lkotlin\/jvm\/functions\/Function0\<Lkotlin\/Unit;\>;)V\lDebug Info:\l  0x0007\-0x001b: r1 'ex' Ljava\/lang\/Throwable;\l  0x0001\-0x001b: r0 '$i$f$callCancelHandlerSafely' I\l  \-1 \-0x001b: r6 'block' Lkotlin\/jvm\/functions\/Function0;\l}"];
MethodNode -> Node_11;Node_11 -> Node_13;
Node_13 -> Node_19;
Node_19 -> Node_14;
Node_19 -> Node_16;
Node_14 -> Node_20;
Node_14 -> Node_21;
Node_20 -> Node_16;
Node_21 -> Node_12;
Node_16 -> Node_17;
Node_17 -> Node_18;
Node_18 -> Node_12;
}

